plugins/modules/cloud/google/gcp_storage_aws_transfer_job.py

```diff
@@ -0,0 +1,272 @@
+
+
+DOCUMENTATION = '''
+---
+module: gcp_storage_aws_transfer_job
+version_added: 2.10
+short_description: Creates GCP Transfer Jobs between AWS S3 and GCP Storage.
+description: This module will create GCP Storage Transfer Jobs which transfers AWS S3 Bucket objects into the specified
+             GCP Bucket. If there is an existing Transfer Job with the same Description it will delete it and create a new job.
+             If you include the parameter aws_s3_bucket_prefix, it will transfer that prefix only, otherwise it will transfer all
+             objects in the bucket. The Transfer Jobs will only pull changed or new objects when they execute.
+
+options:
+    project_id:
+        description:
+            - GCP account project id.
+        required: true
+        type: str
+    scheduled_start_date_utc:
+        description:
+            - Start date of transfer job in YYYY-MM-DD format.
+            - For details on how transfer job schedules work see https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs#schedule.
+        required: true
+        type: str
+    scheduled_end_date_utc:
+        description:
+            - End date of transfer job in YYYY-MM-DD format .
+        required: true
+        type: str
+    scheduled_start_time_utc:
+        description:
+            - Transfer job start time in HH:SS format.
+        required: true
+        type: str
+    gcp_storage_bucket:
+        description:
+            - The GCP storage bucket to transfer objects into.
+        required: true
+        type: str
+    service_account_file:
+        description:
+            - GCP credentails file.
+        required: true
+        type: str
+    aws_s3_bucket:
+        description:
+            - AWS S3 bucket which contains the source objects.
+        required: true
+        type: str
+    aws_s3_bucket_prefix:
+        description:
+            - Prefix within the S3 bucket which contains the source objects.
+        required: false
+        type: str
+    description:
+        description:
+            - This is used as a unique name for the transfer job. If a job with the same description exists, it is replaced, or deleted depending on state.
+        required: true
+        type: str
+    state:
+        description:
+            - Whether to create/update or delete a transfer job. Options are present and absent.
+        required: true
+        type: str
+    aws_access_key:
+        description:
+            - The AWS IAM account access key used by the transfer job to retreive S3 objects.
+        required: true
+        type: str
+    aws_secret_key:
+        description:
+            - The AWS IAM account access key secret used by the transfer job to retreive S3 objects.
+        required: true
+        type: str
+
+author:
+    - Chanaka Samarajeewa (@csamarajeewa)
+'''
+
+EXAMPLES = '''
+
+# Create a new job / replace existing job
+
+- name: Create Bucket Transfer Job
+  gcp_storage_aws_transfer_job:
```

Comment:
Always use FQCNs (Fully Qualified Collection Names) in examples:
```suggestion
  community.general.gcp_storage_aws_transfer_job:
```

Reply:
@felixfontein , would you like me to close this request and create two separate ones?

Reply:
You can also remove one of the new modules from this PR, and create a new PR for the other.